specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/dataVersion.json

```diff
@@ -0,0 +1,383 @@
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "machineLearningServices.json#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/DataVersionNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DataVersionVersionsParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Data Version entity to create or update.",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "201": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/DataVersionResource"
+            }
+          },
+          "default": {
+            "description": "Error",
+            "schema": {
+              "$ref": "machineLearningServices.json#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DataVersionNameParameter": {
+      "in": "path",
+      "name": "name",
+      "type": "string",
+      "description": "Data Version Name identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    },
+    "DataVersionVersionsParameter": {
+      "in": "path",
+      "name": "version",
+      "type": "string",
+      "description": "Data Version Version identifier.",
+      "required": true,
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "definitions": {
+    "DataVersionResource": {
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DataVersion"
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "machineLearningServices.json#/definitions/ResourceBase"
+        }
+      ]
+    },
+    "DataVersionResourcePaginatedResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DataVersionResource"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "DataVersion": {
+      "description": "Details of an Data Version.",
+      "type": "object",
+      "properties": {
+        "datasetType": {
+          "$ref": "#/definitions/DatasetType"
+        },
+        "datastoreId": {
```

Comment:
Is this a resource ID of some other resource?  If so, what permissions does someone need on it to use it in a dataVersion resource and how do you validate that?

Reply:
This is a resourceID of a datastore resource. Permissions are handled by Datastore and should be resolved by the time this ID reaches DataVersion. We validate it by using the error handling built-into the ResourceArmScope parsing.